### PR TITLE
feat(mobile): editing wheel results bb-344

### PR DESCRIPTION
**Pull request by Olha-Katran**
Create scores editing form using slider with gradient color
Add nested route from "wheel" to "Edit my wheel results"
Add style to page "Wheel"
Add placeholder place for checkbox form
Add getScore slice

For now - use mock data, without save editing changes

screenshot:
  

Video: 

https://github.com/user-attachments/assets/fd984e51-173d-4391-a65e-979a72f35ccf

### CODE REVIEW

File: apps/mobile/src/libs/components/white-background-wrapper/white-background-wrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import { View } from "~/libs/components/components";
+import { globalStyles } from "~/libs/styles/styles";
+
+import { styles } from "./styles";
+
+type Properties = {
+	children: React.ReactNode;
+};
+
+const WhiteBackgroundWrapper: React.FC<Properties> = ({ children }) => {
```
Comment (revenets) [inline apps/mobile/src/libs/components/white-background-wrapper/white-background-wrapper.tsx:12]: Don't think that we actually need this, Can we just pass the `style` prop to ScreenWrapper?

File: apps/mobile/src/screens/wheel/wheel.tsx [hunk 2]
```diff
@@ -1,23 +1,55 @@
...
+			<ScreenWrapper>
+				<View style={[globalStyles.m16]}>
+					<Text preset="subheading" weight="bold">
+						My Wheel Results
+					</Text>
+					<Text
+						preset="tag"
+						style={[globalStyles.pv4, globalStyles.ph12, styles.date]}
+					>
+						5 Aug 2024, Tuesday
```
Comment (revenets) [inline apps/mobile/src/screens/wheel/wheel.tsx:40]: I think we can use actual date here, not mocked value

File: apps/mobile/src/screens/edit-wheel-results/components/scores-edit-form/scores-edit-form.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	const handleSliderChange = useCallback(
+		(categoryId: number, value: number) => {
+			setScores(
+				scores.map((item) =>
+					item.categoryId === categoryId ? { ...item, score: value } : item,
+				),
+			);
+		},
+		[scores],
+	);
```
Comment (revenets) [inline apps/mobile/src/screens/edit-wheel-results/components/scores-edit-form/scores-edit-form.tsx:46]: We should use react-hook-form for this form as well

**Review by revenets — CHANGES_REQUESTED** [review]

File: apps/mobile/src/screens/edit-wheel-results/components/scores-edit-form/scores-edit-form.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+const ScoresEditForm: React.FC<Properties> = ({ data }) => {
+	const dispatch = useAppDispatch();
+	const [scores, setScores] = useState<SliderData[]>(data);
+
+	useEffect(() => {
+		void dispatch(quizActions.getScores());
+	}, [dispatch]);
+
+	const handleSaveChange = useCallback(() => {
+		//Todo implement edit scores
```
Comment (fshabanov) [inline apps/mobile/src/screens/edit-wheel-results/components/scores-edit-form/scores-edit-form.tsx:34]: The Backend is already merged, implement this as well

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/mobile/src/libs/components/screen-wrapper/screen-wrapper.tsx [hunk 5]
```diff
@@ -25,7 +28,9 @@ const ScreenWrapper: React.FC<Properties> = ({
 		paddingTop: edges.includes("top") ? insets.top : ZERO_PADDING,
 	};
 
-	return <View style={[globalStyles.flex1, paddingStyle]}>{children}</View>;
+	return (
+		<View style={[globalStyles.flex1, paddingStyle, style]}>{children}</View>
+	);
 };
```
Comment (Shcracoziabra) [inline apps/mobile/src/libs/components/screen-wrapper/screen-wrapper.tsx:35]: Its a good idea to add style to the wrapper!
And what about  adding a Direction enum for directions mentioned in this component?
```ts
const Direction = {
	BOTTOM: "bottom",
	LEFT: "left",
	RIGHT: "right",
	TOP: "top",
} as const;
```

**Review by Shcracoziabra — APPROVED** [review]

File: apps/mobile/src/libs/components/screen-wrapper/screen-wrapper.tsx
Comment (Shcracoziabra) [inline apps/mobile/src/libs/components/screen-wrapper/screen-wrapper.tsx:1]: Its a good idea to add style to the wrapper!
And what about  adding a Direction enum for directions mentioned in this component?
```ts
const Direction = {
	BOTTOM: "bottom",
	LEFT: "left",
	RIGHT: "right",
	TOP: "top",
} as const;
```

**Review by Shcracoziabra — APPROVED** [review]

File: apps/mobile/src/libs/helpers/helpers.ts [hunk 6]
```diff
@@ -2,6 +2,7 @@ export { checkIfAndroid, checkIfIos } from "./check-platform/check-platform";
 export { createRadioButtonProperties } from "./create-radio-button-properties/create-radio-button.properties";
```
Comment (Shcracoziabra) [inline apps/mobile/src/libs/helpers/helpers.ts:2]:
```suggestion
export { createRadioButtonProperties } from "./create-radio-button-properties/create-radio-button-properties.helper";
```

File: apps/mobile/src/libs/components/slider-content/slider-content.tsx [hunk 7]
```diff
@@ -1,34 +1,53 @@
...
 type Properties = {
-	color: keyof typeof colorToGradientColors;
+	id: number;
 	label: string;
+	onValueChange: (categoryId: number, value: number) => void;
+	value: number;
 };
 
 const MAX_SLIDER_VALUE = 10;
 const MIN_SLIDER_VALUE = 0;
```
Comment (Shcracoziabra) [inline apps/mobile/src/libs/components/slider-content/slider-content.tsx:15]: How about creating enum?
```ts
const SliderValue = {
  MAX: 10,
  MIN: 0,
 } as const;
 ```

File: apps/mobile/src/screens/wheel/styles.ts [hunk 8]
```diff
@@ -0,0 +1,18 @@
+import { StyleSheet } from "react-native";
+
+import { BaseColor } from "~/libs/enums/enums";
+
+const styles = StyleSheet.create({
+	container: {
+		display: "flex",
```
Comment (Shcracoziabra) [inline apps/mobile/src/screens/wheel/styles.ts:7]: `display: flex ` is default for React Native layout, this prop is redundant.

**Pull request closed without merging**